lib/compress.js

```diff
@@ -176,6 +176,21 @@ merge(Compressor.prototype, {
         }
     };
 
+    // we shouldn't compress (1,func)(something) to
+    // func(something) because that changes the meaning of
+    // the func (becomes lexical instead of global).
+    function check_function_call(parent, orig, val) {
```

Comment:
I'd prefer it be named something like `fix_lexical_scope`  or `fix_expression_scope` over `check_function_call` because it better conveys its action. It's not so much as a check as a fix up.